pkg/services/ngalert/notifier/channels/alertmanager.go

```diff
@@ -34,18 +33,21 @@ func NewAlertmanagerConfig(config *NotificationChannelConfig, fn GetDecryptedVal
 	if urlStr == "" {
 		return nil, errors.New("could not find url property in settings")
 	}
-	var urls []*url.URL
-	for _, uS := range strings.Split(urlStr, ",") {
+
+	urlParts := strings.Split(urlStr, ",")
+	urls := make([]*url.URL, 0, len(urlParts))
+
+	for _, uS := range urlParts {
 		uS = strings.TrimSpace(uS)
 		if uS == "" {
 			continue
 		}
 		uS = strings.TrimSuffix(uS, "/") + "/api/v1/alerts"
-		url, err := url.Parse(uS)
+		u, err := url.Parse(uS)
```

Comment:
here name collides with "url" package and variable is local so i renamed it to "u", but if it important i can go back this change, it not related to PR in general